server/fishtest/rundb.py

```diff
@@ -603,15 +603,22 @@ def calc_itp(self, run):
             itp = 1
         elif itp > 500:
             itp = 500
-        itp *= math.sqrt(
-            estimate_game_duration(run["args"]["tc"]) / estimate_game_duration("10+0.1")
-        )
-        itp *= math.sqrt(run["args"]["threads"])
-        if "sprt" not in run["args"]:
-            itp *= 0.5
-        else:
+
+        # Base TP derived from power law of TC relative to STC
+        tc_ratio = run["args"]["threads"] * estimate_game_duration(run["args"]["tc"]) \
+                                          / estimate_game_duration("10+0.1")
+        # Discount longer test TP, but don't boost shorter tests
+        if tc_ratio > 1:
+            # LTC/STC tc_ratio = 6, target latency ratio = 3/2,
+            # => LTC base tp = 4 => log(4)/log(6) ~ 0.774
+            itp *= tc_ratio**0.774
+
+        # TP bonus derived from LLR (or min-bonus for nonSPRTs)
+        llr = -2.94
+        if "sprt" in run["args"]:
             llr = run["args"]["sprt"].get("llr", 0)
-            itp *= (5 + llr) / 5
+        itp *= (llr + 8) / 8 # max/min bonus 1.37/0.63
```

Comment:
I would like to go in the different direction, but even more cores on the tests with high llr. That will improve time to answer for strong patches. I suggest `itp *= (llr + 4) / 4`
`

Reply:
I have mixed feelings about this.

1) I have indeed been thinking for some weeks about implementing some extra bonus for the most promising tests, as candirufish and I had been doing manually;

2) More recently, I tried a couple weeks ago to set my most garbage tests to low TP to let them finish while letting more resources go to other, less-garbage tests; however, the end result turned out to be that I had the same dozen tests languishing at the bottom of my page for multiple weeks while other users only ever looked at my testcount, not overall TP. In other words, I've reached the conclusion that too low TP for bad tests only increases the clutter in the test list, so I'm against increasing the malus for bad tests, as implied by `+4/4`. A minimum TP of 0.63 is already low enough (and higher than the current 0.41, and I certainly think going below 0.41 would be even worse).

3) Regardless of 1) and 2), the considerable change in LTC thruput suggested by just tweaking the power law will go a long way to getting the LTC passers going quicker, without needing to give them any extra bonus at all.

I think we should be conservative and see how the power law tweak feels, and then later we can add an extra bonus for the best tests if we still want to. That said, I can include such extra bonus code in this PR here and now for consideration. But again, just the power law tweak will be a considerable shift in LTC latency.

Reply:
I agree with the proposal of do incremental changes, the code can be tested on DEV only for bugs, the effect on the queue must be tested on PROD.